functorch/_src/eager_transforms.py

```diff
@@ -643,6 +644,9 @@ def assert_non_empty_list_of_tensors(output, api, argname):
 
 
 def safe_unpack_dual(dual, strict):
+    if not isinstance(dual, torch.Tensor):
+        raise RuntimeError(f"Expected tensors, got unsupported type {type(dual)}")
```

Comment:
nit: Is this only called in jvp? Maybe we should be a bit more descriptive about where these tensors came from (e.g. "jvp(f, args, tangents): expected f(*args) to return only tensors, got ..."

Reply:
Thinking more about that IMO it would be better either introduce additional arg e.g. `api: str` to specify from where it comes or catch that elsewhere. There is a similar issue with `_undo_create_differentiable` which is used by multiple functions and error message is not always clear...

Reply:
Maybe file an issue for the future?